src/engine/blkstore/blkstore.hpp

```diff
@@ -336,15 +339,20 @@ class BlkStore {
     BlkAllocStatus alloc_blk(const uint32_t size, blk_alloc_hints& hints, std::vector< BlkId >& out_blkid) {
         // Allocate a block from the device manager
         assert(size % m_pagesz == 0);
-        blk_count_t nblks{static_cast< blk_count_t >(size / m_pagesz)};
+        // Using a 16-bit value (blk_count_t) for page counts can lead to overflow issues when managing large storage
+        // devices. With a page size of 4096 bytes, a 16-bit value limits the addressable storage to 256 MB (65,536
+        // pages * 4096 bytes/page).
+        uint32_t nblks{static_cast< uint32_t >(size / m_pagesz)};
         if (nblks <= BlkId::max_blks_in_op()) {
-            return (m_vdev.alloc_blk(nblks, hints, out_blkid));
+            auto max_val = std::numeric_limits< blk_count_t >::max();
+            HS_DBG_ASSERT_LE(nblks, max_val, "nblks must be less than {}", max_val);
```

Comment:
nit:  do we still this assert since you've added static assert at blk.h? I mean the static assert at blk.h can make sure this assert always true, right? Also can we just use one line: `HS_DBG_ASSERT_LE(nblks, std::numeric_limits< blk_count_t >::max(), "nblks must be less than {}", max_val);`, although compiler might do the same for optimization.  I am fine either way.

Reply:
I remove it